plugins/web/opentelemetry-plugin-react-load/src/reactLoad.ts

```diff
@@ -0,0 +1,195 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { BasePlugin, isWrapped } from '@opentelemetry/core';
+import * as shimmer from 'shimmer';
+import { AttributeNames } from './enums/AttributeNames';
+import { VERSION } from './version';
+import * as React from 'react';
+
+/**
+ * This class represents a react lifecycle plugin
+ */
+export class ReactLoad extends BasePlugin<unknown> {
+  readonly component: string = 'react-load';
+  readonly version: string = '1';
+  moduleName = this.component;
+  protected _config!: api.PluginConfig;
+  private _reactComponent: React.Component[];
+  private _parentSpanMap: WeakMap<React.Component, api.Span | undefined>;
+
+  /**
+   * @param reactComponent
+   */
+  constructor(reactComponent: any[]) {
+    super('@opentelemetry/plugin-react-load', VERSION);
+    this._reactComponent = reactComponent.map((item: any) => {
+      return item.prototype;
+    });
+    this._parentSpanMap = new WeakMap<React.Component, api.Span | undefined>();
+  }
+
+   /**
+   * Creates a new span as a child of the current parent span. 
+   * If parent span is undefined, just the child is created.
+   * @param react React component currently being instrumented
+   * @param name name of span
+   */
+  private _createSpanWithParent(
+    react: React.Component,
+    name: string
+  ): api.Span | undefined {
+    return this._tracer.startSpan(name, {
+      attributes: {
+        [AttributeNames.COMPONENT]: this.moduleName,
+      },
+      parent: this._getParentSpan(react),
+    });
+  }
+
+   /**
+   * Creates a new span
+   * @param name name of span
+   */
+  private _createSpan(name: string): api.Span | undefined {
+    return this._tracer.startSpan(name, {
+      attributes: {
+        [AttributeNames.COMPONENT]: this.moduleName,
+      },
+    });
+  }
+
+   /**
+   * Returns the current parent span. If one doesn't exist, 
+   * this function creates a parent span
+   * @param react React component parent span belongs to.
+   */
+  private _getParentSpan(react: React.Component): api.Span | undefined {
+    let span: api.Span | undefined = this._parentSpanMap.get(react);
+    if (!span) {
+      span = this._createSpan('parent');
+      this._parentSpanMap.set(react, span);
+    }
+    return span;
+  }
+
+  /**
+   * Patches the render lifecycle method
+   */
+  private _patchRender() {
+    return (original: () => React.ReactNode): (() => React.ReactNode) => {
```

Comment:
Minor suggestion: what do you think about declaring these react methods as type? ([example](https://github.com/thgao/opentelemetry-js/blob/master/packages/opentelemetry-plugin-xml-http-request/src/types.ts) [usage](https://github.com/thgao/opentelemetry-js/blob/master/packages/opentelemetry-plugin-xml-http-request/src/xhr.ts#L316)). This might help with readability and help others and provide clarity as well.

Reply:
That sounds like it would be clearer since what the actual signature of the functions are not super important to the code, so I agree it'd be better to abstract this more. Thanks!